lib/algos/list/delimiterMaster.js

```diff
@@ -165,14 +170,54 @@ class DelimiterMaster extends Delimiter {
         return super.filter(obj);
     }
 
+    allowDelimiterRangeSkip(key) {
+        if (!this.prefixKeySeen) {
+            return true;
+        }
+        // prefix = prefix, key = prefix/. Can skip since key will be part of commonPrefixes.
+        if (key[key.length - 1] === this.delimiter && key.length > this.prefix.length && key.startsWith(this.prefix)) {
```

Comment:
To be checked but I believe we can remove the two last tests as they are already checked earlier in the function (that the key starts with the prefix), and if it equals the prefix it's already handled as well.

```suggestion
        if (key[key.length - 1] === this.delimiter) {
```

Reply:
removing this will actually allow prefix/ to be skipped. This is why we need to check that the key is of the form prefix/X

Reply:
Actually, we can remove the first test then because it's already checked before calling this function at https://github.com/scality/Arsenal/pull/2011/files#diff-9e520dd5b6e739ad1fba611bf9fdffc713fbaf71ed390ad15565a5263578a1f6R206

```suggestion
        if (key.length > this.prefix.length && key.startsWith(this.prefix)) {
```